### PR TITLE
Guide to use minio-java with clojure.

### LINKED ISSUE #695: [RELEASE] console v0.22.4 doc impacts
Console [v0.22.4](https://github.com/minio/console/releases/tag/v0.22.4) impacts the docs:

- [x] Adds an option to the confirmation box when deleting file versions to bypass governance [PR #2536](https://github.com/minio/console/pull/2536)
- [x] Removes GKE integration [PR #2552](https://github.com/minio/console/pull/2552)

### PR DESCRIPTION
**Pull request by aarushiarya**
Fixes #695. Step-by-step guide to use minio-java with clojure, including example.

### CODE REVIEW

File: docs/Clojure.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+## Getting started with minio-java
+
+First, download minio-java.
+```
+git clone git@github.com:minio/minio-java.git
+```
+
+Compile a jar
+By running this command in minio-java directory -
```
Comment (nitisht) [inline docs/Clojure.md]: This should be one line

File: docs/Clojure.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+)
+
+(defn -main
+  "I don't do a whole lot ... yet."
+  [& args]
+  (println "Hello, World!")
+    (println "Using minio-java")
+    (.putObject (new io.minio.MinioClient "https://play.minio.io:9000" "Q3AM3UQ867SPQQA43P2F" "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG") "testbucketcloj"  "objectname" "/path/to/testFile")
+)
+```
```
Comment (harshavardhana) [inline docs/Clojure.md]: Add a new line

File: docs/Clojure.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+First, download minio-java.
+```
+git clone git@github.com:minio/minio-java.git
+```
+
+Compile a jar by running this command in minio-java directory - 
+```
+./gradlew build
+```
```
Comment (nitisht) [inline docs/Clojure.md]: Is it mandatory to build `minio-java` from source? I think we can point to http://repo1.maven.org/maven2/io/minio/minio/ for users to simply download the jar. For advanced users (who want to build from source) we can add this link https://github.com/minio/minio-java/blob/master/CONTRIBUTING.md#contributors-guide

File: docs/Clojure.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+### 1. Install the jar
+Create a directory in the project.
+```
+mkdir maven_repository
+```
+
+Add local jars to this repository.
+```
+mvn install:install-file -Dfile=minio-5.0.0-DEV-all.jar -DartifactId=minio -Dversion=5.0.0 -DgroupId=minio -Dpackaging=jar -DlocalRepositoryPath=maven_repository DcreateChecksum=true
+```
```
Comment (nitisht) [inline docs/Clojure.md]: Is a step missing here? I get this

```
nitish@minio:~/dev/clojure-minio-java/clojproject$ mvn install:install-file -Dfile=minio-5.0.0-DEV-all.jar -DartifactId=minio -Dversion=5.0.0 -DgroupId=minio -Dpackaging=jar -DlocalRepositoryPath=maven_repository DcreateChecksum=true
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.120 s
[INFO] Finished at: 2018-08-05T15:33:07+05:30
[INFO] Final Memory: 6M/147M
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/home/nitish/dev/clojure-minio-java/clojproject). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
```

File: docs/Clojure.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+Go to project directory and update installed dependencies for the project.
+```
+lein deps
+```
+
+### 2. Download from maven
+Go to `maven.org` and search `minio`. 
+Download the lastest version and add dependency in the `project.clj`.
+
```
Comment (nitisht) [inline docs/Clojure.md]: Are steps 1 and 2 both needed? or any one of these.

Comment (aarushiarya) [reply docs/Clojure.md]: Any one.

File: docs/Clojure.md (hunk 4 above)
Comment (aarushiarya) [reply docs/Clojure.md]: Removed this part.

File: docs/Clojure.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+```
+
+### PutObject
+```
+(ns clojproject.core
+ (import io.minio.MinioClient)
+)
+
+(defn -main
+  "I don't do a whole lot ... yet."
```
Comment (harshavardhana) [inline docs/Clojure.md]: We don't need to write custom sentences like these, we should write what the `main` function does.

File: docs/Clojure.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+```
+/.m2/repository/io/minio/minio/4.0.2/minio-4.0.2.jar
+```
+
+Relaunch repl for Clojure and use minio-java.
+
+## Test
+Use classes and methods from the imported class in your project.
+
+### Make Bucket
```
Comment (harshavardhana) [inline docs/Clojure.md]: This heading should be the same as the next example heading

File: docs/Clojure.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+```
+(ns clojproject.core
+ (import io.minio.MinioClient)
+)
+
+(defn -main
+  "I don't do a whole lot ... yet."
```
Comment (harshavardhana) [inline docs/Clojure.md]: We don't need to write custom sentences like these, we should write what the `main` function does.

File: docs/Clojure.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+Run `lein deps`.
+
+Check `lein classpath` to see if minio.jar is present.
+```
+/.m2/repository/io/minio/minio/4.0.2/minio-4.0.2.jar
+```
+
+Relaunch repl for Clojure and use minio-java.
+
+## Test
```
Comment (harshavardhana) [inline docs/Clojure.md]: Can we restructure this document to look like our rest of the cookbook guides? with Step 1, 2, 3 style?

Comment (aarushiarya) [reply docs/Clojure.md]: Sure.

File: docs/Clojure.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+## Clojure Project
+Following are the steps involved foe using minio-java with Clojure : 
+
+### 1. Setup
+Create a new Clojure project using Leiningen.
+```
+lein new <templateName> <projectName>
+```
+
+```
```
Comment (nitisht) [inline docs/Clojure.md]: This feels incomplete, please add a line here like `For example` or similar

File: docs/Clojure.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+lein new app clojproject
+```
+
+Navigate to `project.clj` look at `:dependencies` section, listing what project requires in order to load, compile and run.
+```
```
Comment (nitisht) [inline docs/Clojure.md]: This line is not clear - please elaborate on what is `dependencies` section and how is `listing` happening

File: docs/Clojure.md (hunk 10 above)
Comment (aarushiarya) [reply docs/Clojure.md]: Okay.

**Pull request merged**